### PR TITLE
[DEV APPROVED] Refactor stamp duty info table

### PR DESCRIPTION
**Pull request by margOnline**
This pr refactors the stamp duty 'how calculated' table to reduce duplication.

### CODE REVIEW

File: spec/helpers/stamp_duties_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+
+module MortgageCalculator
+  describe StampDutiesHelper do
+    describe '#band' do
+      context 'when there is a lower and an upper limit' do
+        it 'returns a currency range with both limits formatted' do
+          expect(band(0, 125000) ).to eq "£0 - £125,000"
```
Comment (Tr4pSt3R) [inline spec/helpers/stamp_duties_helper_spec.rb]: `expect(band(0, 125000) ).to eq "£0 - £125,000"` --> `expect(band(0, 125000)).to eq("£0 - £125,000")`

File: spec/helpers/stamp_duties_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        end
+      end
+
+      context 'for the maximum band' do
+        it 'returns text for the maximum band' do
+          expect(band(1500001, nil) ).to eq "Over £1.5 million"
```
Comment (Tr4pSt3R) [inline spec/helpers/stamp_duties_helper_spec.rb]: `expect(band(1500001, nil) ).to eq "Over £1.5 million"` --> `expect(band(1500001, nil)).to eq('Over £1.5 million')`

File: features/stamp_duty_information_table.feature [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Feature: Stamp Duty Information Table
+In order to understand how my stamp duty is calculated
+As a home buyer
+I want to see what percentages apply to the purchase price
+
+Scenario: As a home buyer
+  Given I visit the Stamp Duty page
+  And I select 'isFTB' as my buyer type
```
Comment (Tr4pSt3R) [inline features/stamp_duty_information_table.feature]: `And I select 'isFTB' as my buyer type` --> `When I am buying a home for the first time`.

The reason for this suggestion is that the implementation behind the scenes does not need to be part of story-telling. This is just a suggestion though.

File: features/step_definitions/stamp_duty.rb [hunk 4]
```diff
@@ -74,3 +80,14 @@
 Then(/^I see the Welsh stamp duty calculator$/) do
   expect(@stamp_duty.h1.first).to have_content('Cyfrifiannell treth stamp')
 end
+
+Then(/^I should see the stamp duty percentages for first time buyers as:$/) do |table|
```
Comment (Tr4pSt3R) [inline features/step_definitions/stamp_duty.rb]: Excellente!

File: app/helpers/mortgage_calculator/stamp_duties_helper.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    private
+    def maximum(num)
+      "#{I18n.t('stamp_duty.table.over')} £#{num/1_000_000.to_f} #{I18n.t('stamp_duty.table.million')}"
+    end
+
+    def formatted_currency(num)
+      number_to_currency(num, precision: 0)
+    end
+  end
+end
```
Comment (tomas-stefano) [inline app/helpers/mortgage_calculator/stamp_duties_helper.rb]: Final newline missing

File: app/helpers/mortgage_calculator/stamp_duties_helper.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+module MortgageCalculator
+  module StampDutiesHelper
+    def band(num1, num2)
+      return maximum(num1 - 1) if num2.nil?
+      "#{formatted_currency(num1)} - #{formatted_currency(num2)}"
+    end
+
+    private
+    def maximum(num)
+      "#{I18n.t('stamp_duty.table.over')} £#{num/1_000_000.to_f} #{I18n.t('stamp_duty.table.million')}"
```
Comment (tomas-stefano) [inline app/helpers/mortgage_calculator/stamp_duties_helper.rb]: Can we make this one locale and pass as an argument the num/1_000_000.to_f?

File: app/models/mortgage_calculator/stamp_duty.rb [hunk 7]
```diff
@@ -17,6 +17,13 @@ def self.i18n_scope
       1000000000 => 12
     }
 
+    SDLT_RATES = [
+      {lower: 0,         upper: 125_000,   first_home: 0,  additional_or_buy_to_let: 3},
+      {lower: 125_001,   upper: 250_000,   first_home: 2,  additional_or_buy_to_let: 5},
+      {lower: 250_001,   upper: 925_000,   first_home: 5,  additional_or_buy_to_let: 8},
+      {lower: 925_001,   upper: 1_500_000, first_home: 10, additional_or_buy_to_let: 13},
+      {lower: 1_500_001, upper: nil,       first_home: 12, additional_or_buy_to_let: 15}
+    ]
```
Comment (tomas-stefano) [inline app/models/mortgage_calculator/stamp_duty.rb]: Don't forget to freeze objects assigned to constants.

File: features/step_definitions/stamp_duty.rb [hunk 8]
```diff
@@ -14,6 +14,12 @@
   @stamp_duty.load
 end
 
+Given("I select {string} as my buyer type") do |buyer_type|
+  @stamp_duty.buyer_type_options.select do |option|
+    option['value'] == buyer_type
+  end.first.click
```
Comment (tomas-stefano) [inline features/step_definitions/stamp_duty.rb]: Can we change the `select {}.first` to `find`?

**Comment by margOnline** [comment]
@tomas-stefano @Tr4pSt3R all comments addressed.

**Comment by tomas-stefano** [comment]
👍

**Pull request merged**